core/src/main/java/org/apache/calcite/sql/validate/SqlAbstractConformance.java

```diff
@@ -144,4 +146,8 @@ public abstract class SqlAbstractConformance implements SqlConformance {
   @Override public boolean allowLenientCoercion() {
     return SqlConformanceEnum.DEFAULT.allowLenientCoercion();
   }
+
+  @Override public RoundingMode roundingMode() {
+    return SqlConformanceEnum.DEFAULT.roundingMode();
```

Comment:
I think rounding should be part of the type system

Reply:
I agree.

I commented elsewhere that including `SqlConformance` in `RexExecutor` made me feel uneasy. `SqlConformance` is about what is allowed in valid SQL.

Reply:
@julianhyde @mihaibudiu I will move the `roundingMode` to `RelDataTypeSystem`. Thanks for the advice.